Makefile

```diff
@@ -1,10 +1,20 @@
+ARCH=
+CC=gcc
 CFLAGS=-O2 -Wall
 DLLINKFLAGS=-Wl,--kill-at -static-libgcc -shared
 
+ifeq ($(shell uname -m),x86_64)
+ifeq ($(ARCH),i686)
+CC = PATH=/mingw32/bin:$$PATH i686-w64-mingw32-gcc -march=i686
+endif
+endif
+
```

Comment:
> FYI: For the MSI (coming on a separate branch), I will need a
> conditional to compile with the appropriate architecture of gcc. But
> it'll be simpler han the one shown here (remove the outer conditional
> and actually assign `ARCH`: `ARCH=$(shell uname -m)`).

Actually, this conditional should not be necessary. The 32-bit installer
can *only* be built in the 32-bit SDK, otherwise the bundled MSYS2 runtime
will target the wrong architecture. Likewise for 64-bit. So: `gcc` will
already compile for the appropriate architecture.